src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -118,24 +118,15 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
           implementsTypes = superinterfaces
         }
 
-        if (!extendsTypes.isEmpty()) {
-          codeWriter.emit(" extends")
-          var firstType = true
-          for (type in extendsTypes) {
-            if (!firstType) codeWriter.emit(",")
-            codeWriter.emit(" %T", type)
-            firstType = false
-          }
+        if(!extendsTypes.isEmpty() || !implementsTypes.isEmpty()) {
+          codeWriter.emit(":")
         }
 
-        if (!implementsTypes.isEmpty()) {
-          codeWriter.emit(" implements")
-          var firstType = true
-          for (type in implementsTypes) {
-            if (!firstType) codeWriter.emit(",")
-            codeWriter.emit(" %T", type)
-            firstType = false
-          }
+        var firstType = true
+        (extendsTypes + implementsTypes).forEach {
+          if (!firstType) codeWriter.emit(",")
```

Comment:
Also – I think we want a space before the ":" to match examples on kotlinlang.org.